test/PoolManager.t.sol

```diff
@@ -0,0 +1,284 @@
+        );
+
+        vm.stopPrank();
+
+        vm.startPrank(deployer);
+        // Unhappy path Nº2 - Trying to create a Pool with a token that is not whitelisted.
+        vm.expectRevert(abi.encodeWithSignature("PoolManager_TokenNotWhitelisted()"));
+        poolManager.createPool(
+            address(token1),
+            depositFeeRecipient,
+            baseRateTokensPerBlock,
+            depositFee,
+            rewardsMultiplierBlocks,
+            rewardsMultipliers,
+            withdrawFeeBlocks,
+            withdrawFees
+        );
+        poolManager.addWhitelistedToken(address(token1));
+
+        // Happy path - Being the Owner and the token is whitelisted.
+        poolManager.createPool(
+            address(token1),
+            depositFeeRecipient,
+            baseRateTokensPerBlock,
+            depositFee,
+            rewardsMultiplierBlocks,
+            rewardsMultipliers,
+            withdrawFeeBlocks,
+            withdrawFees
+        );
+        assertEq(poolManager.getPools().length, 1);
+        address poolAddressToken1_ = poolManager.getPools()[0];
+        //assertTrue(poolManager.getPools()[0].supportsInterface(type(IPool).interfaceId));
+
+        vm.stopPrank();
+
+        vm.startPrank(address(poolManager));
+
+        assertTrue(boosterPack.getWhitelistedAddrBP(poolAddressToken1_));
+
+        vm.stopPrank();
+    }
+
+    function test_createMultiplePools() public {
+        vm.startPrank(deployer);
+
+        poolManager.addWhitelistedToken(address(token1));
+        poolManager.addWhitelistedToken(address(token2));
+
+        // Happy path - Being the Owner and the token is whitelisted.
+        poolManager.createPool(
+            address(token1),
+            depositFeeRecipient,
+            baseRateTokensPerBlock,
+            depositFee,
+            rewardsMultiplierBlocks,
+            rewardsMultipliers,
+            withdrawFeeBlocks,
+            withdrawFees
+        );
+        poolManager.createPool(
+            address(token2),
+            depositFeeRecipient,
+            baseRateTokensPerBlock,
+            depositFee,
+            rewardsMultiplierBlocks,
+            rewardsMultipliers,
+            withdrawFeeBlocks,
+            withdrawFees
+        );
+        address poolAddressToken1_ = poolManager.getPools()[0];
+        address poolAddressToken2_ = poolManager.getPools()[1];
+
+        // IERC165 registerInterface vs mapping to store addresses of new Pools in PoolFactory (as Maple)
+        //assertTrue(poolManager.getPools()[0].supportsInterface(type(IPool).interfaceId));
+        //assertTrue(poolManager.getPools()[1].supportsInterface(type(IPool).interfaceId));
+
+        vm.stopPrank();
+
+        vm.startPrank(address(poolManager));
+
+        assertTrue(boosterPack.getWhitelistedAddrBP(poolAddressToken1_));
+        assertTrue(boosterPack.getWhitelistedAddrBP(poolAddressToken2_));
+
+        vm.stopPrank();
+    }
+
+    function test_claimRewards() public {
```

Comment:
info: it would be best here to mock the pool.claimRewards() inside poolManager.claimRewards(). A full test like this one should be under integration tests and not unit tests.